playground/CosmosEndToEnd/CosmosEndToEnd.AppHost/CosmosEndToEnd.AppHost.csproj

```diff
@@ -9,8 +9,9 @@
   </PropertyGroup>
 
   <ItemGroup>
-    <ProjectReference Include="..\..\..\src\Aspire.Hosting.Azure\Aspire.Hosting.Azure.csproj" />
-    <ProjectReference Include="..\..\..\src\Aspire.Hosting\Aspire.Hosting.csproj" />
+    <ProjectReference Include="..\..\..\src\Aspire.Hosting.Azure\Aspire.Hosting.Azure.csproj" AspireHostLibrary="true" />
```

Comment:
NIT: Not really a suggestion, but I'm not sure I love the name of the metadata. From a user's POV it is not obvious what this means until you understand the context. I wonder if we should instead use something more explicit, like `AddReferenceToAssembly` or something that is more obvious.

Reply:
I agree that I'm not super happy about this name, but I haven't found one I like better yet.

The metadata does more than just take away the reference to the assembly, it is really the switch between "is this an app that you want to orchestrate?" or "is this a library that has AppHost extension logic you want to use?".

Another idea is to flip it to be `AspireServiceApp=false`. We already have `ServiceNameOverride` if you want to specify the name of the `Projects` class. Setting `AspireServiceApp=false` would mean "this isn't an application, so reference it like a normal ProjectReference".

Reply:
super nit: Prefixing the name with `Is*` would make it more readable.

Reply:
> super nit: Prefixing the name with `Is*` would make it more readable.

Agreed. I think initially I didn't add `Is*` because the other ProjectReference metadata didn't follow this pattern:

* ReferenceOutputAssembly 
* SkipGetTargetFrameworkProperties 
* ExcludeAssets 

But these are more "should" and not "is", so I think it is a different scenario.

So should we have:

* `IsAspireHostLibrary=true`
* -or-
* `IsAspireServiceApp=false`
?

I think I'm leaning towards `IsAspireHostLibrary=true` because the naming aligns with `IsAspireHost=true` for the whole project.

Reply:
I don't really know yet all the special stuff done for these projects, so not sure what a `ServiceApp` vs `Library` would mean.

Reply:
ServiceApp is literally any type of project which represents a microservice that you want to launch via the AppHost. This means the AppHost will orchestrate those types of projects.

The AppHost, while it is in reality an orchestrator, in the end is just a C# project, so you could think of scenarios where people want to add project references to libraries too (which aren't really microservices). For example, imagine a company that has several Aspire apps, and they want to share some of the logic they have in their AppHost projects. For this, they would build a shared library, which AppHost projects would all reference. This library, is not really a service, but instead just a library used by the AppHost.

Reply:
So:
- `ServiceApp` - special project that the AppHost project will *use* in some way, maybe post-process etc.
- Other case - simple *library* projects, and AppHost will reference the assemblies from these, and use them
- Wasn't there another case of referencing projects that are simply meant as a build dependency, and should not affect AppHost project's build itself?

Reply:
> ServiceApp - special project that the AppHost project will use in some way, maybe post-process etc.

It isn't a "special" project. But ServiceApps are just regular ASP.NET apps - like a web api, or blazor server app. They can also be "Worker" apps, if they don't need to expose an HTTP endpoint.

The AppHost's Program.cs files uses these apps by calling `builder.AppProject<ServiceApp>("serviceName")`.

> Wasn't there another case of referencing projects that are simply meant as a build dependency, and should not affect AppHost project's build itself?

I don't know of this case. In my head there are mainly 2 cases: ServiceApps and "normal" ProjectReferences to libraries.

Reply:
> I don't know of this case. In my head there are mainly 2 cases: ServiceApps and "normal" ProjectReferences to libraries.

My misunderstanding then.

I would vote for `IsAspireServiceApp=false` because we are changing the default `ProjectReference` to mean a ServiceApp reference which once you know it, makes sense. IOW, `AppHost` project references `ServiceApp`s by default. In that case for any other project references an opt-out of that default would make sense.

Reply:
Discussing this more with @DamianEdwards, we decided to go with the name:

* `IsAspireProjectResource=false`
    * Setting this to false opts you into being an "AppHost library", you don't get "Projects" code generated, and you start referencing the project

to align with the "Resource" terminology in the app model.

We can then do some more renames:

* `ServiceNameOverride` => `AspireProjectMetadataTypeName`
* `IServiceMetadata` => `IProjectMetadata`